### DISCUSSION TITLE
Help the developer of guinget

### DISCUSSION THREAD

**Discussion by BUGProtectionAssociation** [Devolutions/UniGetUI#480] [General]
WinGet works great. There's another winget GUI program. But its developers seem to be having trouble lately
The[guinget](https://github.com/DrewNaylor/guinget) It seems that it has not changed for a long time. Developers may not have the time and energy to solve some problems. Maybe we can invite him to join us in perfecting wingetui and Help him solve the problem.

**Comment by marticliment** [comment]
oh, I saw guinget and the way it works is completely different from wingetui. while wingetui uses the winget.exe command, guinget implements the whole cli itself.

honestly, I don't see how we could help them and how they could help us, but if you have any ideas make sure to post them here

**Reply by DrewNaylor** [reply]
(I was just curious to see if anything had been mentioned in this repo because this project had previously been mentioned in mine, so that's why I'm here right now.)

Yeah, some winget commands don't quite work well enough for my needs, so I implemented my own method of doing certain things like getting the list of packages, but to clarify, the packages do get sent to winget.exe to be installed/upgraded/uninstalled once the user is ready to do so.

Edit: oh, and thank you for using "they" to refer to me as I don't have my pronouns specified anywhere (because I don't want to) and that helps to normalize trans people's pronouns by not assuming what they are even though I'm cis.

**Reply by BUGProtectionAssociation** [reply]
Your guinget has some problems synchronizing. It's causing it to be very slow, but I really can't find out what the cause is. WingetUI is better optimized for speed, but should not be as efficient as executing native commands directly (instead of calling winget.exe indirectly)

**Reply by DrewNaylor** [reply]
The slowness in loading the package list is due to the way I was loading it in the versions that are available to download from Releases, but the current code doesn't have certain fixes integrated from the latest release which needs to be done for v0.4. If this is referring to extracting the ZIP files, that's an issue with the .NET file extraction library just being inefficient from what I understand. Could switch to DotNetZip, but I don't know if that's safe with regards to licensing as it's under the Ms-PL. Have been looking into trying to integrate 7-Zip into the extraction process beyond the option available in the `Experimental` tab of the `Options` window, but still haven't taken the time to hook up the cancelation thing to allow users to stop extraction as well as displaying the extraction progress and handling errors. (Edit: Found `RecursiveExtractor` from Microsoft under the MIT license and maybe it can help make extracting faster without having to figure out how to get 7z.exe's output? https://github.com/microsoft/RecursiveExtractor) (Edit 2: maybe I should just do multi-threading when extracting? Never tried doing that before, though, and I don't know if it would still be possible to display the current file being extracted with threads. Got the idea from a post on SO that mentioned doing that with the `ZipArchive` class and `ZipArchiveEntry` or something.)

I don't quite understand the part about executing native commands directly instead of calling winget.exe indirectly.
